mmv1/third_party/terraform/transport/config.go.tmpl

```diff
@@ -435,11 +435,19 @@ func (c *Config) LoadAndValidate(ctx context.Context) error {
 	}
 
 	c.TokenSource = tokenSource
-
 	cleanCtx := context.WithValue(ctx, oauth2.HTTPClient, cleanhttp.DefaultClient())
+	clientOptions := []option.ClientOption{option.WithTokenSource(tokenSource)}
+
+	// 'X-Goog-User-Project' gets set by a transport layer within transport.NewHTTPClient
+	// setting explicitly here to override the setting of it by underlying utility function.
```

Comment:
I read the linked comment but don't quite understand what we're solving for here- could you reexplain it, sorry?

Reply:
Basically, the default transport layer will set this header if `GOOGLE_CLOUD_QUOTA_PROJECT` is defined. Since the transport layer is defined below our logging transport their definition takes precedent.  

This is effectively solving for https://github.com/hashicorp/terraform-provider-google/issues/17882 .

Reply:
Thank you! Maybe something like this?

"The client libraries allow setting the GOOGLE_CLOUD_QUOTA_PROJECT environment variable directly, which unintentionally takes precedence over provider settings. Ensure that provider settings take precedence by applying to the client library's client directly."

Reply:
added ! Thanks for the rephrase! :)